org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/markers/SonarLintMarkerResolutionGenerator.java

```diff
@@ -39,11 +43,28 @@ public boolean hasResolutions(final IMarker marker) {
 
   @Override
   public IMarkerResolution[] getResolutions(final IMarker marker) {
+    List<IMarkerResolution> resolutions = new ArrayList<>();
+    resolutions.add(new ShowRuleDescriptionMarkerResolver(marker));
+
     if (hasResolutions(marker)) {
-      return new IMarkerResolution[] {new ShowIssueFlowsMarkerResolver(marker), new ShowRuleDescriptionMarkerResolver(marker)};
-    } else {
-      return new IMarkerResolution[] {new ShowRuleDescriptionMarkerResolver(marker)};
+      resolutions.add(new ShowIssueFlowsMarkerResolver(marker));
+    }
+
+    if (isStandalone(marker)) {
+      resolutions.add(new DeactivateRuleMarkerResolver(marker));
     }
+
+    // note: the display order seems independent from the order in this array
+    return resolutions.toArray(new IMarkerResolution[resolutions.size()]);
+  }
+
+  private boolean isStandalone(IMarker marker) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "isStandalone" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.eclipse%3Asonarlint-eclipse-parent&pullRequest=145&issues=AWKzShfmool-HGyLJW7K&open=AWKzShfmool-HGyLJW7K)

[ISSUE_KEY]: AWKzShfmool-HGyLJW7K